### ISSUE TITLE
Lost allocations leave running tasks that are accumulating

### ISSUE THREAD

**Issue by FRosner** [FRosner/cluster-broccoli#280]
### Problem

Related to the change introduced in #248, I found a problem where "lost" allocations would leave "running" tasks behind. The UI was accumulating with those tasks. Unfortunately I didn't take a screenshot and now I'm not able to reproduce it again.

The problem is that I don't know what I did to Nomad to make it lose allocations :(. If this behaviour of the UI is consistent to the behaviour of nomad and lost allocations leave running tasks behind then we can't do anything.

```
nomad status my-jupyter
ID            = my-jupyter
Name          = my-jupyter
Type          = service
Priority      = 50
Datacenters   = dc1
Status        = running
Periodic      = false
Parameterized = false

Summary
Task Group  Queued  Starting  Running  Failed  Complete  Lost
jupyter     0       0         1        0       0         34

Allocations
ID        Eval ID   Node ID   Task Group  Desired  Status   Created At
d30881f4  ea0f519b  d3314d98  jupyter     run      running  08/18/17 14:52:40 CEST
9739bde3  d8a5bb7e  d3314d98  jupyter     stop     lost     08/18/17 14:52:22 CEST
324ea9cd  f3cf80e8  d3314d98  jupyter     stop     lost     08/18/17 14:52:08 CEST
0cd05950  271e63b4  d3314d98  jupyter     stop     lost     08/18/17 14:51:50 CEST
4e759ca2  7307055a  d3314d98  jupyter     stop     lost     08/18/17 14:51:37 CEST
e87e169d  0c0be795  d3314d98  jupyter     stop     lost     08/18/17 14:51:18 CEST
05bb320a  23a9669c  d3314d98  jupyter     stop     lost     08/18/17 14:51:05 CEST
7c840966  91f296c1  d3314d98  jupyter     stop     lost     08/18/17 14:50:52 CEST
25589761  971b967d  d3314d98  jupyter     stop     lost     08/18/17 14:50:36 CEST
b7804611  79fa594d  d3314d98  jupyter     stop     lost     08/18/17 14:50:24 CEST
13944c30  7fc6b822  d3314d98  jupyter     stop     lost     08/18/17 14:50:11 CEST
95d06de9  f64e0774  d3314d98  jupyter     stop     lost     08/18/17 14:49:57 CEST
f1b2d935  5bc88f33  d3314d98  jupyter     stop     lost     08/18/17 14:49:38 CEST
79d34d6c  a0541d01  d3314d98  jupyter     stop     lost     08/18/17 14:49:22 CEST
7aa50f5e  6fe5dcae  d3314d98  jupyter     stop     lost     08/18/17 14:49:04 CEST
d5243bed  216a5ab5  d3314d98  jupyter     stop     lost     08/18/17 14:48:50 CEST
103ed4a5  aa89bfdf  d3314d98  jupyter     stop     lost     08/18/17 14:48:35 CEST
926f5a43  ba3ba139  d3314d98  jupyter     stop     lost     08/18/17 14:48:22 CEST
08280480  c27ad124  d3314d98  jupyter     stop     lost     08/18/17 14:48:09 CEST
10cc84ff  08932629  d3314d98  jupyter     stop     lost     08/18/17 14:47:49 CEST
a6c9429f  3011860b  d3314d98  jupyter     stop     lost     08/18/17 14:47:35 CEST
d1995502  121e711f  d3314d98  jupyter     stop     lost     08/18/17 14:47:20 CEST
259eb23a  094238be  d3314d98  jupyter     stop     lost     08/18/17 14:47:06 CEST
f37aace2  aa4e9d44  d3314d98  jupyter     stop     lost     08/18/17 14:46:55 CEST
c49b38ab  4908c625  d3314d98  jupyter     stop     lost     08/18/17 14:46:35 CEST
963213b7  05dea9ec  d3314d98  jupyter     stop     lost     08/18/17 14:46:19 CEST
3f78301a  d876d22a  d3314d98  jupyter     stop     lost     08/18/17 14:46:02 CEST
d746c7d4  29cc8709  d3314d98  jupyter     stop     lost     08/18/17 14:45:47 CEST
c2881235  c5596a4b  d3314d98  jupyter     stop     lost     08/18/17 14:45:30 CEST
6970a19f  b8ddd69b  d3314d98  jupyter     stop     lost     08/18/17 14:45:13 CEST
22ee3283  e4fe107e  d3314d98  jupyter     stop     lost     08/18/17 14:44:56 CEST
3cb3d225  147563d6  d3314d98  jupyter     stop     lost     08/18/17 14:44:37 CEST
ed66288e  f5a397e5  d3314d98  jupyter     stop     lost     08/18/17 14:44:20 CEST
e01450bc  03c7ce4b  d3314d98  jupyter     stop     lost     08/18/17 14:44:09 CEST
693f3c78  e02d455a  d3314d98  jupyter     stop     lost     08/18/17 14:43:59 CEST
```

Is there anything we can do right now or should we just keep this in mind?

**Comment by swsnr** [comment]
@FRosner Did you run `alloc-status` on any of the lost allocations?  I bet they'd have listed the actual task in the allocation as "running"… what you're seeing is the allocation status but #248 shows the task state (we need to get down to the level of tasks to show logs with Nomad's API).

**Comment by FRosner** [comment]
No I didn't unfortunately. Turned out that for some reason after suspending my laptop and keeping nomad running for a few weeks, it somehow lost track of the docker wait commands, maybe because I was shutting down docker in the meantime. Anyway I was not able to reproduce, so nevermind.

**Comment by swsnr** [comment]
@FRosner I was able to "reproduce" the issue; not deliberately though, it just happened when I tried to use the built-in Jupyter template.

I used `nomad alloc-status` and the Nomad API, and indeed: While Nomad has lost the allocation it still reports the task of that allocation as running, as seen in `TaskEvents` of the allocation JSON.

I don't know why Nomad behaves this way… I see that the built-in Jupyter template seems to use docker, but docker for mac was running in my case.

Even more concerning, Nomad couldn't get to a stable state again: After loosing the first allocation it'd constantly try to spawn new allocations only to loose them immediately again, accumulating more and more lost allocations with tasks reported to be running, although I don't think that any task was running at all—at least nothing appeared in `docker ps`.  

I don't know, this doesn't make me very confident about Nomad.
